packages/effect/src/Schema.ts

```diff
@@ -820,7 +820,17 @@ export const TemplateLiteralParser = <Params extends array_.NonEmptyReadonlyArra
     }
   }
   const from = TemplateLiteral(...encodedSchemas as any)
-  const re = AST.getTemplateLiteralCapturingRegExp(from.ast as AST.TemplateLiteral)
```

Comment:
The issue is that this type coercion doesn't tell the truth. The `TemplateLiteral` constructor has a code path where it returns an `AST.Literal` instead of an `AST.TemplateLiteral`.